### PR TITLE
Add minHeight option to sizeToFit modifier

### PR DESCRIPTION
**Pull request by daltojohnso**
This lets us set a reasonable minHeight value on a `TippyPopover` that uses the `sizeToFit` modifier so that the popover doesn't become too small. Pass the option object as the value of the prop `sizeToFit={{ minHeight: 500 }}`. Right now, can only be a number value because we use it for comparisons.

This is one option. It favors trying to preserve the original placement. Beyond the min-height, it flips the placement, but will fall back to the original if the space meets the minimum requirement (the minHeight value). If there's not enough height above or below, it defaults back to the original `placement` and overflows.

Another option I could implement is a sort of "aggressive" flip which will flip to whichever area has more space, regardless of the original placement.

https://user-images.githubusercontent.com/13057258/166988160-692256e3-2c1c-4986-bebc-18083cfaaa00.mov

### CODE REVIEW

File: frontend/src/metabase/components/Popover/TippyPopover.tsx [hunk 1]
```diff
@@ -21,12 +22,13 @@ export interface ITippyPopoverProps extends TippyProps {
...
   lazy?: boolean;
   flip?: boolean;
-  sizeToFit?: boolean;
+  sizeToFit?: boolean | SizeToFitOptions;
   onClose?: () => void;
 }
 
 const PAGE_PADDING = 10;
 const OFFSET: [number, number] = [0, 5];
+const SIZE_TO_FIT_MIN_HEIGHT = 200;
```
Comment (WiNloSt) [inline frontend/src/metabase/components/Popover/TippyPopover.tsx:31]: Not sure why linter doesn't warn/throw an error about this line. I don't think it's being used in this file.

**Review by WiNloSt — APPROVED** [review]
Tested and works pretty well. Could be great if we could get this into `master` so I could complete https://github.com/metabase/metabase/issues/15925 and chip away one more popover migration for https://github.com/metabase/metabase/issues/18951

File: frontend/src/metabase/components/Popover/TippyPopover.tsx (hunk 1 above)
Comment (daltojohnso) [reply frontend/src/metabase/components/Popover/TippyPopover.tsx]: Odd, not sure why the linter doesn't catch this, but I've removed all the unused things in https://github.com/metabase/metabase/pull/22438/commits/52418c40d3bcff1c31dc8d390358fb4a38e467d4

Comment (WiNloSt) [reply frontend/src/metabase/components/Popover/TippyPopover.tsx]: [I think I've seen this before (Slack)](https://metaboat.slack.com/archives/C505ZNNH4/p1651672938330139). Not sure if it's relavant.

**Pull request merged**